core/storage/src/lib.rs

```diff
@@ -122,6 +124,30 @@ macro_rules! ensure_get {
     }};
 }
 
+macro_rules! impl_get_keys_by_schemas_for {
+    ($self_: ident, $schema: ty, $vec: expr, $key_prefix: expr, $local_ibc_prefix: expr) => {{
+        let prepare_iter = $self_.adapter.prepare_iter::<$schema, _>($key_prefix)?;
```

Comment:
`prepare_iter` will get all pathes which starts with key_prefix?

Reply:
`prepare_iter` is a build-in function, it will return all keys which suit the schema struct like `ClientTypeSchema`, then the function `get_paths_by_prefix` will  filte the result second time.

Reply:
`prepare_iter` won`t get path which starts with key_prefix, it just get all paths.